odoo/addons/base/models/ir_ui_view.py

```diff
@@ -2628,7 +2628,7 @@ def _get_view_cache_key(self, view_id=None, view_type='form', **options):
     @api.model
     @tools.conditional(
         'xml' not in config['dev_mode'],
-        tools.ormcache('self._get_view_cache_key(view_id, view_type, **options)'),
+        tools.ormcache_context('self._get_view_cache_key(view_id, view_type, **options)', keys=('force_email',)),
```

Comment:
Mmmh ?

Reply:
I’m not fond of it either but it is to fix an existing problem discovered during the tests. 
The problem is that _get_view_cache uses _get_view which result depends on the context variable  “force_email” in its redefinition in res_partner. As the cache doesn’t depends on it, the result is random (depending of the context variable of the first call). A better solution would be to integrate that in the options parameter which is covered in the cache dependencies but force_email is used at a lot of places so it would not be easy …

Reply:
Why you don't use `form_view_ref` in the caller view?

Don't change base model for a specific addon feature.
nb: If the key cache need any change, the change must be in `_get_view_cache_key` (overwrite, not in base).

Reply:
Thank you for pointing me to the right solution. I have overridden _get_view_cache_key to add the force_email context to the cache key as it would be hard to remove the use of that context variable which has an impact on the returned view.